File: core/src/ru/myitschool/dcrawler/MyGdxGame.java
package ru.myitschool.dcrawler;

import com.badlogic.gdx.*;
import com.badlogic.gdx.graphics.g2d.BitmapFont;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import ru.myitschool.dcrawler.encounters.Encounter;
import ru.myitschool.dcrawler.entities.Enemy;
import ru.myitschool.dcrawler.entities.Entity;
import ru.myitschool.dcrawler.event.EntityEvent;
import ru.myitschool.dcrawler.event.EventAction;
import ru.myitschool.dcrawler.event.EventController;
import ru.myitschool.dcrawler.event.EventListener;
import ru.myitschool.dcrawler.screens.MainMenuScreen;
import ru.myitschool.dcrawler.tiles.DungeonTile;

import java.util.Arrays;

public class MyGdxGame extends Game
{
    private MainMenuScreen mainMenuScreen;
    private boolean debugInfo = false;

    protected BitmapFont font;
    protected SpriteBatch batch;

    private static InputMultiplexer input;

    @Override
    public void create() {
        font = new BitmapFont();
        batch = new SpriteBatch();
        EventListener.initListeners();
        EventController.addActionAfterEvent(new EventAction() {
            @Override
            protected void act(String eventCode) {
                if (Arrays.asList(EntityEvent.eventCodes).contains(eventCode)){
                    Entity.getPlayingEntities().removeIf(entity -> !entity.isAlive());
                }
            }
        });
        DungeonTile.initTiles();
        Encounter.initEncouters();
        Enemy.createCRTable();
        input = new InputMultiplexer();
        input.addProcessor(new InputAdapter());

        //this.gameScreen = new GameScreen(this);
        this.mainMenuScreen = new MainMenuScreen();
        this.setScreen(this.mainMenuScreen);
        input.addProcessor(new InputAdapter(){
            @Override
            public boolean keyUp(int keycode) {
                System.out.println(keycode);
                if (keycode == Input.Keys.GRAVE){
                    debugInfo = !debugInfo;
                }
                if (keycode == Input.Keys.R){
                    resize(1920, 1080);
                }
                return false;
            }
        });
        Gdx.input.setInputProcessor(input);
    }

    public static void setScreenInput(InputProcessor inputProcessor){
        input.getProcessors().set(0, inputProcessor);
    }

    @Override
    public void render() {
        super.render();
        if (debugInfo) {
            float h = Gdx.graphics.getHeight();
            batch.begin();
            font.draw(batch, "FPS: " + Gdx.graphics.getFramesPerSecond(), 10, h - 20);
            //font.draw(charBatch, "camera: " + camera.position, 10, h - 50); TODO draw in screen class
            batch.end();
        }
    }

    @Override
    public void resize(int width, int height) {
        super.resize(width, height);
    }

    @Override
    public void pause() {
        // TODO Auto-generated method stub
    }

    @Override
    public void resume() {
        // TODO Auto-generated method stub
    }

    @Override
    public void dispose() {
        // TODO Auto-generated method stub
    }
}

File: core/src/com/wolg_vlad/dcrawler/MyGdxGame.kt
package com.wolg_vlad.dcrawler

import com.badlogic.gdx.*
import com.badlogic.gdx.graphics.g2d.BitmapFont
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.wolg_vlad.dcrawler.encounters.Encounter
import com.wolg_vlad.dcrawler.entities.Enemy
import com.wolg_vlad.dcrawler.entities.Entity
import com.wolg_vlad.dcrawler.event.EntityEvent
import com.wolg_vlad.dcrawler.event.EventAction
import com.wolg_vlad.dcrawler.event.EventController
import com.wolg_vlad.dcrawler.event.EventListener
import com.wolg_vlad.dcrawler.ui.screens.MainMenuScreen
import com.wolg_vlad.dcrawler.ui.tiles.DungeonTile
import java.util.*

class MyGdxGame : Game() {
    private var mainMenuScreen: MainMenuScreen? = null
    private var debugInfo = false
    protected var font: BitmapFont? = null
    protected var batch: SpriteBatch? = null
    override fun create() {
        font = BitmapFont()
        batch = SpriteBatch()
        EventListener.initListeners()
        EventController.addActionAfterEvent(object : EventAction() {
            override fun act(eventCode: String) {
                if (Arrays.asList<String>(*EntityEvent.eventCodes).contains(eventCode)) {
                    Entity.playingEntities.removeIf { entity: Entity -> !entity.isAlive }
                }
            }
        })
        DungeonTile.initTiles()
        Encounter.initEncouters()
        Enemy.createCRTable()
        input = InputMultiplexer()
        input!!.addProcessor(InputAdapter())
        //this.gameScreen = new GameScreen(this);
        mainMenuScreen = MainMenuScreen()
        setScreen(mainMenuScreen)
        input!!.addProcessor(object : InputAdapter() {
            override fun keyUp(keycode: Int): Boolean {
                println(keycode)
                if (keycode == Input.Keys.GRAVE) {
                    debugInfo = !debugInfo
                }
                if (keycode == Input.Keys.R) {
                    resize(1920, 1080)
                }
                return false
            }
        })
        Gdx.input.inputProcessor = input
    }

    override fun render() {
        super.render()
        if (debugInfo) {
            val h = Gdx.graphics.height.toFloat()
            batch!!.begin()
            font!!.draw(batch, "FPS: " + Gdx.graphics.framesPerSecond, 10f, h - 20)
            //font.draw(charBatch, "camera: " + camera.position, 10, h - 50); TODO draw in screen class
            batch!!.end()
        }
    }

    override fun resize(width: Int, height: Int) {
        super.resize(width, height)
    }

    override fun pause() { // TODO Auto-generated method stub
    }

    override fun resume() { // TODO Auto-generated method stub
    }

    override fun dispose() { // TODO Auto-generated method stub
    }

    companion object {
        private var input: InputMultiplexer? = null
        fun setScreenInput(inputProcessor: InputProcessor?) {
            input!!.processors[0] = inputProcessor
        }
    }
}